packages/stellar_client/lib/src/client.dart

```diff
@@ -0,0 +1,440 @@
+        body: jsonEncode({'transaction': fundedTransaction.toXdr()}),
+      );
+
+      print(response.body);
+    } catch (error) {
+      throw Exception('Something went wrong! $error');
+    }
+  }
+
+  Future<List<Map<String, dynamic>>> getTransactions() async {
+    Page<OperationResponse> payments = await _sdk.payments
+        .forAccount(accountId)
+        .order(RequestBuilderOrder.DESC)
+        .execute();
+    List<Map<String, dynamic>> transactionDetails = [];
+
+    if (payments.records != null && payments.records!.isNotEmpty) {
+      for (OperationResponse response in payments.records!) {
+        if (response is PaymentOperationResponse) {
+          var details = _handlePaymentOperationResponse(response);
+          transactionDetails.add({'type': 'Payment', 'details': details});
+        } else if (response is CreateAccountOperationResponse) {
+          var details = _handleCreateAccountOperationResponse(response);
+          transactionDetails.add({'type': 'CreateAccount', 'details': details});
+        } else if (response is PathPaymentStrictReceiveOperationResponse) {
+          var details =
+              _handlePathPaymentStrictReceiveOperationResponse(response);
+          transactionDetails
+              .add({'type': 'PathPaymentStrictReceive', 'details': details});
+        } else {
+          print("Unhandled operation type: ${response.runtimeType}");
+        }
+      }
+    } else {
+      print("No payment records found.");
+    }
+
+    return transactionDetails;
+  }
+
+  PathPaymentStrictReceiveOperationDetails
+      _handlePathPaymentStrictReceiveOperationResponse(
+          PathPaymentStrictReceiveOperationResponse response) {
+    return PathPaymentStrictReceiveOperationDetails(
+      from: response.from!,
+      to: response.to!,
+      sourceAmount: double.parse(response.sourceAmount!),
+      sourceAssetCode: response.sourceAssetCode ?? 'XLM',
+      destinationAmount: double.parse(response.amount!),
+      destinationAssetCode: response.assetCode!,
+    );
+  }
+
+  PaymentOperationDetails _handlePaymentOperationResponse(
+      PaymentOperationResponse response) {
+    return PaymentOperationDetails(
+      from: response.from!.accountId,
+      to: response.to!.accountId,
+      amount: double.parse(response.amount!),
+      assetCode: response.assetCode ?? 'XLM',
+      isSuccessful: response.transactionSuccessful!,
+    );
+  }
+
+  CreateAccountOperationDetails _handleCreateAccountOperationResponse(
+      CreateAccountOperationResponse response) {
+    return CreateAccountOperationDetails(
+      account: response.account!,
+      startingBalance: double.parse(response.startingBalance!),
+    );
+  }
+
+  Future<List<Map<String, dynamic>>> getBalance() async {
+    List<Map<String, dynamic>> balancesList = [];
+    AccountResponse account = await _sdk.accounts.account(accountId);
+    for (Balance balance in account.balances) {
+      Map<String, dynamic> balanceMap = {};
```

Comment:
we already know the type here that should have asset code and balance. can we create an interface for it?